clvm-derive/src/helpers.rs

```diff
@@ -33,37 +42,23 @@ pub fn parse_args(attrs: &[Attribute]) -> ClvmDeriveArgs {
                     .unwrap();
 
                 for arg in args {
+                    let existing = repr;
+
                     match arg.to_string().as_str() {
-                        "tuple" => {
-                            if let Some(existing) = repr {
-                                panic!("`tuple` conflicts with `{}`", existing.to_string());
-                            }
-                            repr = Some(Repr::Tuple);
-                        }
-                        "proper_list" => {
-                            if let Some(existing) = repr {
-                                panic!("`proper_list` conflicts with `{}`", existing.to_string());
-                            }
-                            repr = Some(Repr::ProperList);
-                        }
-                        "curried_args" => {
-                            if let Some(existing) = repr {
-                                panic!("`curried_args` conflicts with `{}`", existing.to_string());
-                            }
-                            repr = Some(Repr::CurriedArgs);
-                        }
-                        ident => panic!("unknown argument `{}`", ident),
+                        "tuple" => repr = Some(Repr::Tuple),
+                        "list" => repr = Some(Repr::List),
+                        "curry" => repr = Some(Repr::Curry),
```

Comment:
it looks like you could factor out the `repr = Some(...)` from the match arms to make the whole match expression return the `Repr`